pkg/profiler/profile_writer.go

```diff
@@ -57,22 +58,30 @@ func (fw *FileProfileWriter) Write(_ context.Context, labels model.LabelSet, pro
 
 // RemoteProfileWriter is a profile writer that writes profiles to a remote profile store.
 type RemoteProfileWriter struct {
+	zw                 *gzip.Writer
 	profileStoreClient profilestorepb.ProfileStoreServiceClient
 }
 
 // NewRemoteProfileWriter creates a new RemoteProfileWriter.
 func NewRemoteProfileWriter(profileStoreClient profilestorepb.ProfileStoreServiceClient) *RemoteProfileWriter {
 	return &RemoteProfileWriter{
+		zw:                 gzip.NewWriter(nil),
```

Comment:
`RemoteProfileWriter.Write` is not called concurrently (please correct me if I am wrong).